core/src/main/java/com/linecorp/armeria/client/endpoint/SlowStartAwareEndpointSelectionStrategy.java

```diff
@@ -24,42 +24,63 @@
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
-import com.linecorp.armeria.client.ClientFactory;
 import com.linecorp.armeria.client.ClientRequestContext;
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.client.endpoint.WeightedRoundRobinStrategy.EndpointsAndWeights;
 
-import io.netty.util.concurrent.ScheduledFuture;
-
 /**
- * A weighted round-robin strategy that lamp up the weight gradually from zero to its normal weight value.
+ * A weighted round-robin strategy that ramps up the weight gradually from zero to its normal weight value.
  */
 public final class SlowStartAwareEndpointSelectionStrategy implements EndpointSelectionStrategy {
-    private final EndpointWeighter endpointWeighter;
-    private final ClientFactory clientFactory;
+
+    /**
+     * Creates a {@link SlowStartAwareEndpointSelectionStrategyBuilder}.
+     */
+    public static SlowStartAwareEndpointSelectionStrategyBuilder builder() {
+        return new SlowStartAwareEndpointSelectionStrategyBuilder();
+    }
+
+    /**
+     * Creates a {@link SlowStartAwareEndpointSelectionStrategy}.
+     */
+    public static EndpointSelectionStrategy of(EndpointWeightTransition endpointWeightTransition,
+                                               ScheduledExecutorService executorService,
+                                               Duration slowStartInterval,
+                                               int numberOfSteps) {
```

Comment:
How about reducing the number of parameters of this method by introducing some default values? e.g. `weightTransition` could default to `linear()`. `numerOfSteps` could be `20`.